src/btc.rs

```diff
@@ -116,59 +117,44 @@ pub struct Payload {
 
 #[derive(thiserror::Error, Debug)]
 pub enum PayloadError {
-    #[error("{0}")]
-    AddressError(#[from] bitcoin::address::Error),
-    #[error("invalid witness program size")]
-    InvalidWitnessProgramSize,
-    #[error("witness version {0} not supported yet")]
-    UnsupportedWitnessVersion(WitnessVersion),
     #[error("unrecognized pubkey script")]
     Unrecognized,
 }
 
-impl TryFrom<bitcoin::address::Payload> for Payload {
-    type Error = PayloadError;
-    fn try_from(value: bitcoin::address::Payload) -> Result<Self, Self::Error> {
-        match value {
-            bitcoin::address::Payload::PubkeyHash(h) => Ok(Payload {
-                data: h[..].to_vec(),
+impl Payload {
+    pub fn from_pkscript(pkscript: &[u8]) -> Result<Payload, PayloadError> {
```

Comment:
I suggest changing the type to `Script` for clarity. (But I get why it's not in this commit.)